### PR TITLE
[HOPSWORKS-1054] Reimplement the schemaless metadata API to use the e…

### PR DESCRIPTION
**Pull request by maismail**
…xtended attributes added in HOPS-1312

## Make sure there is no duplicate PR for this issue

* **Please check if the PR meets the following requirements**
- [ ] Adds tests for the submitted changes (for bug fixes & features)
- [x] Passes the tests
- [x] HOPSWORKS JIRA issue has been opened for this PR
- [x] All commits have been squashed down to a single commit

* **Post a link to the associated JIRA issue**
https://logicalclocks.atlassian.net/browse/HOPSWORKS-1054

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the new behavior (if this is a feature change)?**

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:
tests: https://github.com/logicalclocks/hops-testing/pull/466

### CODE REVIEW

File: hopsworks-api/src/main/java/io/hops/hopsworks/api/project/MetadataService.java [hunk 1]
```diff
@@ -585,50 +586,46 @@ private Response mutateMetadata(Users user, String metaObj, MetadataOp op)
...
-  public Response attachSchemalessMetadata(String metaObj) throws DatasetException, MetadataException {
-    processSchemalessMetadata(metaObj, false);
-    return noCacheResponse.getNoCacheResponseBuilder(Response.Status.ACCEPTED).build();
-  }
-
-  @POST
-  @Path("detachSchemaless")
+  
+  @PUT
+  @Path("schemaless/{xattrName}/{path: .+}")
```
Comment (tkakantousis) [inline hopsworks-api/src/main/java/io/hops/hopsworks/api/project/MetadataService.java]: @maismail Please add Swagger ApiOperation annotations to the endpoints. An example is [this](https://github.com/logicalclocks/hopsworks/blob/master/hopsworks-api/src/main/java/io/hops/hopsworks/api/jobs/JobsResource.java#L145).

File: hopsworks-api/src/main/java/io/hops/hopsworks/api/project/MetadataService.java [hunk 2]
```diff
@@ -585,50 +586,46 @@ private Response mutateMetadata(Users user, String metaObj, MetadataOp op)
...
-    String inodePath = jsonObj.getString("path");
-    if (detach) {
-      metadataController.removeSchemaLessMetadata(inodePath);
-      return;
+    if (metaObj.length() > 13500 || xattrName.length() > 255) {
+      throw new MetadataException(RESTCodes.MetadataErrorCode.METADATA_MAX_SIZE_EXCEEDED, Level.FINE);
     }
+    metadataController.addSchemaLessMetadata(user, path, xattrName,
+        metaObj);
+    return noCacheResponse.getNoCacheResponseBuilder(Response.Status.CREATED).build();
```
Comment (tkakantousis) [inline hopsworks-api/src/main/java/io/hops/hopsworks/api/project/MetadataService.java]: @maismail Remove the `nocacheResponse`. Use Response as we do in `JobsResource`

Comment (maismail) [reply hopsworks-api/src/main/java/io/hops/hopsworks/api/project/MetadataService.java]: done

File: hopsworks-api/src/main/java/io/hops/hopsworks/api/project/MetadataService.java (hunk 1 above)
Comment (maismail) [reply hopsworks-api/src/main/java/io/hops/hopsworks/api/project/MetadataService.java]: done

**Pull request merged**